pkg/rook/flexvolume.go

```diff
@@ -0,0 +1,462 @@
+package rook
+
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"io/fs"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kurl/pkg/k8sutil"
+	"github.com/replicatedhq/kurl/pkg/rook/static/flexmigrator"
+	"github.com/ricardomaraschini/plumber"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const (
+	rookCephNamespace                       = "rook-ceph"
+	rookCephMigratorDeploymentName          = "rook-ceph-migrator"
+	rookCephMigratorPodContainerName        = "rook-ceph-migrator"
+	rookCephMigratorDeploymentLabelSelector = "app=rook-ceph-migrator"
+
+	desiredScaleAnnotation = "kurl.sh/rook-flexvolume-to-csi-desired-scale"
+)
+
+// FlexvolumeToCSIOpts are options for the FlexvolumeToCSI function
+type FlexvolumeToCSIOpts struct {
+	// SourceStorageClass is the name of the storage class to migrate from
+	SourceStorageClass string
+	// DestinationStorageClass is the name of the storage class to migrate to
+	DestinationStorageClass string
+	// PVMigratorBinPath is the path to the ceph/pv-migrator binary
+	PVMigratorBinPath string
+	// CephMigratorImage is the image to use for the ceph/pv-migrator container
+	CephMigratorImage string
+}
+
+// Validate will validate options to the FlexvolumeToCSI function and return an error if any are
+// invalid
+func (o FlexvolumeToCSIOpts) Validate() error {
+	if o.SourceStorageClass == "" {
+		return errors.New("source storage class is required")
+	}
+	if o.DestinationStorageClass == "" {
+		return errors.New("destination storage class is required")
+	}
+	if o.PVMigratorBinPath == "" {
+		return errors.New("pv migrator binary path is required")
```

Comment:
Don't we "own" this file ? I mean, aren't we the ones that installed this binary in the node ? If that is the case then why don't we hardcode this for sake of simplicity ?

Reply:
well we sort of do but this binary is under the home directory in our dev environment and the /var/lib/kurl directory in prod.